### PR TITLE
[#2200] keyboard selection only composable for `focus-outline`

### PR DESCRIPTION
**Pull request by aluarius**
Close: #2200 

## Description
- [x] composable for keyboard only focus style conditioning,
- [x] applied to every component that using `focus-outline`,
- [x] related refactoring,
- [x] `focus-outline` applied for `VaTabs`,
- [x] outline offset fix for separate cases.  

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/composables/useKeyboardFocusStyle.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { computed, ExtractPropTypes } from 'vue'
+
+import { useKeyboardOnlyFocus } from './'
+
+export const useKeyboardFocusStyleProps = { disableFocusStyle: { type: Boolean, default: false } }
+
+export function useKeyboardFocusStyle (props: ExtractPropTypes<typeof useKeyboardFocusStyleProps>) {
```
Comment (m0ksem) [inline packages/ui/src/composables/useKeyboardFocusStyle.ts:7]: Can't see any focus style here.

```suggestion
export function useKeyboardFocusable (props: ExtractPropTypes<typeof useKeyboardFocusStyleProps>) {
```

**Review by m0ksem — APPROVED** [review]
How about to overengineer this thing? Just for research.

We might have SCSS mixin:

```scss
@mixin () {
   &--keyboard-focused {
     // make some bruh here
   }
}
```

and 

```html
<div 
  class="va-pricol"
  v-bind="{ ...keyboardBind }" // style and listeners
 />
```

```ts
return {
   ...useKeyboardFocusStyle
}
```

File: packages/ui/src/components/va-icon/VaIcon.vue [hunk 2]
```diff
@@ -80,12 +90,16 @@ export default defineComponent({
       lineHeight: sizeComputed.value,
     }))
 
+    const ariaHiddenComputed = computed(() => attrs.role !== 'button')
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-icon/VaIcon.vue:93]:
```suggestion
    const ariaHiddenComputed = computed(() => attrs.role !== 'button' || attrs.tabindex < 0)
```

File: packages/ui/src/components/va-input/VaInput.vue [hunk 3]
```diff
@@ -34,10 +34,9 @@
       <va-icon
         v-if="canBeCleared"
         role="button"
-        aria-hidden="false"
         aria-label="reset"
-        class="va-input__icons__reset"
         :tabindex="tabIndexComputed"
+        :disable-focus-style="$props.disableFocusClass"
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-input/VaInput.vue:39]: `va-icon` doesn't have this prop
Same in other places

File: packages/ui/src/components/va-tabs/components/VaTab/VaTab.vue [hunk 4]
```diff
@@ -75,7 +83,11 @@ export default defineComponent({
     const hoverState = ref(false)
     const rightSidePosition = ref(0)
     const leftSidePosition = ref(0)
-    const { hasKeyboardFocus, keyboardFocusListeners } = useKeyboardOnlyFocus()
+
+    const { keyboardFocusListeners: keyboardOnlyFocusListeners, hasKeyboardFocus } = useKeyboardOnlyFocus()
+    const { keyboardFocusClass, keyboardFocusListeners } = useKeyboardFocusClass(props, 'va-tab__content')
+    const listeners = { ...keyboardOnlyFocusListeners, ...keyboardFocusListeners }
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-tabs/components/VaTab/VaTab.vue:89]:
```suggestion
    const { keyboardFocusClass, keyboardFocusListeners, hasKeyboardFocus } = useKeyboardFocusClass(props, 'va-tab__content')
    const listeners = keyboardFocusListeners
```

File: packages/ui/src/composables/useKeyboardFocusClass.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+export const useKeyboardFocusClassProps = { disableFocusClass: { type: Boolean, default: false } }
+
+export function useKeyboardFocusClass (props: ExtractPropTypes<typeof useKeyboardFocusClassProps>, parentClass?: string) {
+  const { hasKeyboardFocus, keyboardFocusListeners } = useKeyboardOnlyFocus()
+
+  const hasKeyboardFocusClass = computed(() => !props.disableFocusClass && hasKeyboardFocus.value)
+
+  const keyboardFocusClass = computed(() => ({ [`${parentClass}--keyboard-focus`]: hasKeyboardFocusClass.value }))
+
+  return { keyboardFocusListeners, keyboardFocusClass, hasKeyboardFocusClass }
```
Comment (RVitaly1978) [inline packages/ui/src/composables/useKeyboardFocusClass.ts:14]:
```suggestion
  return { keyboardFocusListeners, keyboardFocusClass, hasKeyboardFocusClass, hasKeyboardFocus }
```

**Review by RVitaly1978 — CHANGES_REQUESTED** [review]

File: packages/ui/src/composables/useKeyboardFocusClass.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import { useKeyboardOnlyFocus } from './'
+
+export const useKeyboardFocusClassProps = { disableFocusClass: { type: Boolean, default: false } }
+
+export function useKeyboardFocusClass (props: ExtractPropTypes<typeof useKeyboardFocusClassProps>, parentClass?: string) {
+  const { hasKeyboardFocus, keyboardFocusListeners } = useKeyboardOnlyFocus()
+
+  const hasKeyboardFocusClass = computed(() => !props.disableFocusClass && hasKeyboardFocus.value)
+
+  const keyboardFocusClass = computed(() => ({ [`${parentClass}--keyboard-focus`]: hasKeyboardFocusClass.value }))
```
[the remark this answers is not in the document]
Comment (aluarius) [reply packages/ui/src/composables/useKeyboardFocusClass.ts:12]: Without passed `parentClass` it won't work anyway. You want to change `undefined` to empty string?

[the remark this answers is not in the document]
Comment (aluarius) [reply packages/ui/src/composables/useKeyboardFocusClass.ts:12]: Changed to empty string. Warning here is not suitable because `parentClass` is optional.

**Review by RVitaly1978 — APPROVED** [review]

**Comment by RVitaly1978** [comment]
But I don't think you should set `outline-offset: -2px` for `va-button` inside components.
We have to rework the components with `overflow: hidden`.

**Comment by aluarius** [comment]
> But I don't think you should set `outline-offset: -2px` for `va-button` inside components. We have to rework the components with `overflow: hidden`.

Yeah, will make a different issue a bit later.

File: packages/ui/src/components/va-rating/components/VaRatingItem/VaRatingItem.vue [hunk 7]
```diff
@@ -101,14 +104,19 @@ export default defineComponent({
 
     watch(hoveredValue, () => emit('hover', hoveredValue.value || RatingValue.EMPTY))
 
+    const { keyboardFocusClass, keyboardFocusListeners } = useKeyboardFocusClass(props, 'va-rating-item')
```
Comment (m0ksem) [inline packages/ui/src/components/va-rating/components/VaRatingItem/VaRatingItem.vue:107]: const componentName = kebabCase(getCurrentInstance()?.name)

**Review by m0ksem — APPROVED** [review]
Still complex.

How about

File: packages/ui/src/components/va-rating/components/VaRatingItem/VaRatingItem.vue (hunk 7 above)
Comment (aluarius) [reply packages/ui/src/components/va-rating/components/VaRatingItem/VaRatingItem.vue]: I've added this option, but it's not universal - some components should pass not root class.

**Pull request merged**